### ISSUE TITLE
[BUG] A child process terminated abruptly

### ISSUE THREAD

**Issue by atus42** [Chia-Network/chia-blockchain#8928]
**Describe the bug**

Lately I faced this situation three times. A chia full node gets stucked at some point, and didn't post proofs nor update the bolckchain. After a clean stop and start, the node can not sync either, in the debug.log file (info level) the following messages appear repeatedly:
```
2021-10-25T17:55:08.426 full_node chia.full_node.full_node: INFO     Start syncing from fork point at 1046419 up to 1048126
2021-10-25T17:55:11.135 full_node chia.full_node.full_node: ERROR    sync from fork point failed err: A child process terminated abruptly, the process pool is not usable anymore
2021-10-25T17:55:11.135 full_node chia.full_node.full_node: INFO     long sync done
```

It repeats seemingly forever, and can not break out of this loop. Stopping chia, and starting again (several times) solves at some point.

After the first occurence of the above error, there is another which may relate:
```
2021-10-25T17:41:33.136 full_node asyncio                 : ERROR    Task was destroyed but it is pending!
task: <Task pending name='Task-1415' coro=<FullNode.sync_from_fork_point.<locals>.fetch_block_batches() running at chia\full_node\full_node.py:804> wait_for=<Future pending cb=[<TaskWakeupMethWrapper object at 0x00000194F8246580>()]> cb=[gather.<locals>._done_callback() at asyncio\tasks.py:766]>
```

I also find this error in the log which may relate:
```
2021-10-25T17:55:40.476 full_node chia.full_node.full_node: INFO     Requesting weight proof from peer 108.29.71.64 up to height 1048131
2021-10-25T17:55:41.022 full_node full_node_server        : INFO     Connection closed: 108.29.71.64, node id: 2f281df46fc85e99242b2f3b7a23fd35bb487d0655a15434a08e63230ea09241
2021-10-25T17:55:41.023 full_node chia.full_node.full_node: INFO     peer disconnected {'host': '108.29.71.64', 'port': 8444}
2021-10-25T17:55:41.025 full_node chia.full_node.full_node: ERROR    Error with syncing: <class 'RuntimeError'>Traceback (most recent call last):
  File "chia\full_node\full_node.py", line 718, in _sync
RuntimeError: Weight proof did not arrive in time from peer: 108.29.71.64
```

**To Reproduce**

There is no known how to reproduce procedure I know of. I just let the node run, and check if it stil 'online'. If not, usually this is the case/problem.

**Expected behavior**
I expect the node to sync without problem at first start.

**Screenshots**

not applicable.

**Desktop**

- OS: windows 10
- OS Version/Flavor: 21H1 19043.1288
- CPU: AMD RYSEN 5

**Additional context**
N/A

**Comment by atus42** [comment]
It happened several times so far. Today I searched the first exception in the debug log files.
The exceptions starts with this:

```
2021-11-03T23:05:17.194 wallet chia.wallet.wallet_node    : INFO     self._close
2021-11-03T23:05:17.195 wallet chia.wallet.wallet_node    : INFO     self._await_closed
2021-11-03T23:05:18.893 wallet wallet_server              : INFO     Connection closed: 127.0.0.1, node id: 1636dc80f42589c7cce04ae2e3bcfb583d1fa784e515e409c888391ee07ef8ed
2021-11-03T23:05:18.894 daemon __main__                   : INFO     Websocket exception. Closing websocket with chia_full_node code = 1006 (connection closed abnormally [internal]), no reason Traceback (most recent call last):
  File "asyncio\windows_events.py", line 457, in finish_recv
OSError: [WinError 64] A megadott hálózatnév már nem érhető el

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "websockets\protocol.py", line 827, in transfer_data
  File "websockets\protocol.py", line 895, in read_message
  File "websockets\protocol.py", line 971, in read_data_frame
  File "websockets\protocol.py", line 1047, in read_frame
  File "websockets\framing.py", line 105, in read
  File "asyncio\streams.py", line 723, in readexactly
  File "asyncio\streams.py", line 517, in _wait_for_data
  File "asyncio\proactor_events.py", line 280, in _loop_reading
  File "asyncio\windows_events.py", line 812, in _poll
  File "asyncio\windows_events.py", line 461, in finish_recv
ConnectionResetError: [WinError 64] A megadott hálózatnév már nem érhető el

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "Lib\site-packages\chia\daemon\server.py", line 193, in safe_handle
  File "websockets\protocol.py", line 439, in __aiter__
  File "websockets\protocol.py", line 509, in recv
  File "websockets\protocol.py", line 803, in ensure_open
websockets.exceptions.ConnectionClosedError: code = 1006 (connection closed abnormally [internal]), no reason

2021-11-03T23:05:18.990 wallet chia.wallet.wallet_blockchain: INFO     Started 10 processes for block validation
```
The phrase "A megadott hálózatnév már nem érhető el" means something like "The given network name is not available any more"
Chia was able to 'recover' from this partly, because I saw harvester and full_node messages in the log:

```
2021-11-03T23:08:44.327 full_node chia.full_node.full_node: INFO     peer disconnected {'host': '95.174.53.236', 'port': 8444}
2021-11-03T23:08:44.464 full_node chia.full_node.full_node: INFO       Finished signage point 39/64: CC: 6e253f04540dbb75f1645fd6808678c38864d7ccb95cf4019d38e0d63b7063a8 RC: dda8e0cb8a2084425b1dd809ab02c7f314492bea3545290fe2d9c4d02640e116 
2021-11-03T23:08:44.478 harvester chia.harvester.harvester: INFO     0 plots were eligible for farming 77b439f3ae... Found 0 proofs. Time: 0.01099 s. Total 145 plots
...
2021-11-03T23:08:51.364 full_node chia.full_node.full_node: INFO       Finished signage point 40/64: CC: cf19e341864e265bd032c5ec782910103abf2dc976c9ca24cb25a998d21a53d0 RC: 486a9222f694ae5400afa0a3567684cb17db1e4fc5cfd8a9c0eafa41545e9fc8 
2021-11-03T23:08:51.441 full_node chia.full_node.full_node: INFO       Finished signage point 41/64: CC: 72183915fdb05d8e2ba0ab5f04df6e55c1e10cd1afdd8bf5d941869c7d80eea6 RC: fa76afc96f9625d25240dd6413ae3dee5249ec763d341d817525537f07d4ff46 
2021-11-03T23:08:51.522 full_node chia.full_node.full_node: INFO       Finished signage point 42/64: CC: ab07ce6bbf99cf8eb3e1aa771fdc213cccfc5e0c2a709d5c6f9ec1747feb01b7 RC: 63b0b6095a0f1b25579a0cd27bf1a4af9fd6426e0cba057f316311bb115cc2b1 
2021-11-03T23:08:51.603 full_node chia.full_node.full_node: INFO       Finished signage point 43/64: CC: 1da29c19aa3d26c9ac45779c670868b805a128ae783344e402920d17b8105c7b RC: 35499dfb0213c80e41fecf30064bf60b6f1ddb96f200b3d8dc21484514b81c44 
```
For some reason, after a while only the "child process terminated abruptly" messages 'win':
```
2021-11-03T23:09:31.158 full_node full_node_server        : ERROR    Exception: A child process terminated abruptly, the process pool is not usable anymore, {'host': '84.120.48.90', 'port': 8444}. Traceback (most recent call last):
  File "chia\server\server.py", line 559, in wrapped_coroutine
  File "chia\full_node\full_node_api.py", line 115, in new_peak
  File "chia\full_node\full_node.py", line 428, in new_peak
  File "chia\full_node\full_node.py", line 362, in short_sync_backtrack
  File "chia\full_node\full_node.py", line 359, in short_sync_backtrack
  File "chia\full_node\full_node.py", line 1309, in respond_block
  File "chia\consensus\blockchain.py", line 644, in pre_validate_blocks_multiprocessing
  File "chia\consensus\multiprocess_validation.py", line 299, in pre_validate_blocks_multiprocessing
  File "asyncio\base_events.py", line 814, in run_in_executor
  File "concurrent\futures\process.py", line 681, in submit
concurrent.futures.process.BrokenProcessPool: A child process terminated abruptly, the process pool is not usable anymore

2021-11-03T23:09:31.160 full_node full_node_server        : ERROR    Exception: A child process terminated abruptly, the process pool is not usable anymore <class 'concurrent.futures.process.BrokenProcessPool'>, closing connection {'host': '84.120.48.90', 'port': 8444}. Traceback (most recent call last):
  File "chia\server\server.py", line 569, in api_call
  File "asyncio\tasks.py", line 442, in wait_for
  File "chia\server\server.py", line 566, in wrapped_coroutine
  File "chia\server\server.py", line 559, in wrapped_coroutine
  File "chia\full_node\full_node_api.py", line 115, in new_peak
  File "chia\full_node\full_node.py", line 428, in new_peak
  File "chia\full_node\full_node.py", line 362, in short_sync_backtrack
  File "chia\full_node\full_node.py", line 359, in short_sync_backtrack
  File "chia\full_node\full_node.py", line 1309, in respond_block
  File "chia\consensus\blockchain.py", line 644, in pre_validate_blocks_multiprocessing
  File "chia\consensus\multiprocess_validation.py", line 299, in pre_validate_blocks_multiprocessing
  File "asyncio\base_events.py", line 814, in run_in_executor
  File "concurrent\futures\process.py", line 681, in submit
concurrent.futures.process.BrokenProcessPool: A child process terminated abruptly, the process pool is not usable anymore
```

The machine has 16G phisical memory and 2x5120MB virtual memory. One of the later is on an nvme card, the other is on a spinning hard drive.

**Comment by atus42** [comment]
With the 1.2.10 version I didn't face this problem so far.
On the other ahnd I don't know the reason. Is it fixed or I didn't face it for some other reason.
Anyway I close this ticet for now, and reopen if it occures again.
